src/utils/propUtils.js

```diff
@@ -10,36 +10,83 @@ import Image from '../elements/Image/Image'
 // ===============================================================
 export const customPropTypes = {
   /**
-   * Ensures children are of a set of types.
-   * Similar to `oneOfType` built-in validator
-   * @param {String[]} allowedTypes Collection of allowed Stardust component types
+   * Ensures children are of a set of types. Matches are made against the component _meta.name property.
+   * @param {String[]} allowedTypes Collection of allowed component types.
    */
   ofComponentTypes: (allowedTypes) => {
     return (props, propName, componentName) => {
-      const { children } = props
-      const disallowed = Children.map(children, child => {
+      if (propName !== 'children') {
+        throw new Error(`ofComponentTypes can only be used on the \`children\` prop, not ${propName}`)
+      }
+      if (!_.isArray(allowedTypes)) {
+        throw new Error([
+          'Invalid argument supplied to ofComponentTypes, expected an instance of array.'
+            ` See ${componentName} prop \`${propName}\`.`,
+        ].join(''))
+      }
+      const disallowed = _.compact(Children.map(props.children, child => {
         return _.includes(allowedTypes, _.get(child, 'type._meta.name')) ? null : child
-      })
-      if (disallowed && disallowed.length !== 0) {
+      }))
+      if (!_.isEmpty(disallowed)) {
         return new Error(
           `\`${componentName}\` should only have children of type \`${allowedTypes}\`.`
         )
       }
     }
   },
+
   /**
-   * Verifies exclusivity of a given prop
-   * @param {string} exclusives property used for exclusivity check
+   * Verifies exclusivity of a given prop.
+   * @param {string[]} exclusives An array of props that cannot be used with this prop.
    */
   mutuallyExclusive: exclusives => {
     return (props, propName, componentName) => {
-      _.forEach(exclusives, exclusiveProp => {
-        if (props[propName] && props[exclusiveProp]) {
-          return new Error(
```

Comment:
Prop type functions must return an error.  This function however was returning an error inside of a forEach.  The validation error was never returned to the prop type function.  It now properly returns the errors.